test/src/unit-capi-query_2.cc

```diff
@@ -3063,3 +3064,70 @@ TEST_CASE_METHOD(
 
   remove_array(array_name);
 }
+
+TEST_CASE_METHOD(
+    Query2Fx,
+    "C API: Test subarray, sparse, set bulk point ranges",
+    "[capi][query_2][sparse][bulk-ranges]") {
+  SECTION("- No serialization") {
+    serialize_ = false;
+  }
+  SECTION("- Serialization") {
+    serialize_ = true;
+  }
+  std::string array_name = "subarray_sparse_default_bulk_ranges";
+  remove_array(array_name);
+  create_sparse_array(array_name);
+
+  // Allocate subarray with an unopened array
+  tiledb_array_t* array;
+  int rc = tiledb_array_alloc(ctx_, array_name.c_str(), &array);
+  CHECK(rc == TILEDB_OK);
+
+  // Open array
+  rc = tiledb_array_open(ctx_, array, TILEDB_READ);
+  CHECK(rc == TILEDB_OK);
+
+  // Create query
+  tiledb_query_t* query = nullptr;
+  rc = tiledb_query_alloc(ctx_, array, TILEDB_READ, &query);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_query_set_layout(ctx_, query, TILEDB_UNORDERED);
+  CHECK(rc == TILEDB_OK);
+
+  // Add ranges
+  uint64_t ranges[] = {1, 3, 7, 10};
+  rc = tiledb_query_add_point_ranges(ctx_, query, 0, ranges, 4);
+  CHECK(rc == TILEDB_OK);
+  rc = tiledb_query_add_point_ranges(ctx_, query, 1, ranges, 4);
+  CHECK(rc == TILEDB_OK);
+
+  // Check that there are four ranges
+  uint64_t range_num;
+  rc = tiledb_query_get_range_num(ctx_, query, 0, &range_num);
+  CHECK(rc == TILEDB_OK);
+  CHECK(range_num == 4);
+
+  // Check ranges
+  for (size_t dim_idx = 0; dim_idx < 1; dim_idx++) {
+    for (size_t idx = 0; idx < 4; idx++) {
+      const void *start, *end, *stride;
+      rc = tiledb_query_get_range(
```

Comment:
There is a CI error for this call on Windows:
` warning C4267: 'argument': conversion from 'size_t' to 'uint32_t', possible loss of data`

Reply:
Updated